aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -0,0 +1,153 @@
+#include <ATen/ATen.h>
+#include <ATen/InitialTensorOptions.h>
+#include <ATen/SparseCsrTensorImpl.h>
+#include <ATen/SparseTensorImpl.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/core/LegacyTypeDispatch.h>
+
+namespace at {
+namespace {
+DeviceType SparseCsrTensorSetToDeviceType(DispatchKeySet key_set) {
+  if (key_set.has(DispatchKey::SparseCsrCPU)) {
+    return kCPU;
+  } else if (key_set.has(DispatchKey::SparseCsrCUDA)) {
+    return kCUDA;
+  } else {
+    AT_ERROR(
+        "Cannot construct SparseTensor with non-sparse tensor type ID ",
+        key_set);
+  }
+}
+} // namespace
+
+SparseCsrTensorImpl::SparseCsrTensorImpl(
+    at::DispatchKeySet key_set,
```

Comment:
Its used by `make_tensor<SparseCsrTensorImpl>()` for setting up a CSR object which can later be resized and reassigned. However (see my comment below), I think it makes sense to not copy the COO approach in this case since we don't have a strong use case for this as of now.